### ISSUE TITLE
false positive: global-variable-not-assigned and invalid-name for global dict

### ISSUE THREAD

**Issue by Avasam** [PyCQA/pylint#5798]
### Bug description

My code snippet, my variables don't need to be assigned to, they're assigned once outside the function then reused.

![image](https://user-images.githubusercontent.com/1350584/153693039-5d9fcd8a-c091-4abe-9716-406fe3114090.png)

```py
__currently_updating_from: dict[ColumnString, datetime] = {}
__currently_updating_to: dict[str, datetime] = {}


@authentication_required
def __do_update_player(current_user: Player, name_or_id: str):
    global __currently_updating_from
    global __currently_updating_to
    now = datetime.now()
    minutes_5 = 5 * 60

    # Check if the current user is already updating someone
    # or if the player to be updated is currently being updated
    if current_user.user_id in __currently_updating_from:
        seconds_left = minutes_5 - (now - __currently_updating_from[current_user.user_id]).total_seconds()
        if seconds_left > 0:
            return {"messageKey": "current_user", "timeLeft": seconds_left}, 409
    if name_or_id in __currently_updating_to:
        seconds_left = minutes_5 - (now - __currently_updating_to[name_or_id]).total_seconds()
        if seconds_left > 0:
            return {"messageKey": "name_or_id", "timeLeft": seconds_left}, 409
    __currently_updating_from[current_user.user_id] = now
    __currently_updating_to[name_or_id] = now

    return __do_update_player_bypass_restrictions(name_or_id, current_user=current_user)
```

### Command used

```shell
pylint --score=n --output-format=colorized $(git ls-files 'backend/*.py')
```

### Pylint output

```shell
************* Module global_scoreboard_api
backend\api\global_scoreboard_api.py:79:4: C0103: Constant name "__currently_updating_from" doesn't conform to UPPER_CASE naming style (invalid-name)
backend\api\global_scoreboard_api.py:79:4: W0602: Using global for '__currently_updating_from' but no assignment is done (global-variable-not-assigned)
backend\api\global_scoreboard_api.py:80:4: C0103: Constant name "__currently_updating_to" doesn't conform to UPPER_CASE naming style (invalid-name)
backend\api\global_scoreboard_api.py:80:4: W0602: Using global for '__currently_updating_to' but no assignment is done (global-variable-not-assigned)
```

### Expected behavior

No warnings

### Pylint version

```shell
pylint 2.12.2
astroid 2.9.3
Python 3.9.6 (tags/v3.9.6:db3ff76, Jun 28 2021, 15:26:21) [MSC v.1929 64 bit (AMD64)]
```

### OS / Environment

Windows 10

**Comment by mbyrnepr2** [comment]
I think this looks like expected behaviour because the dictionaries are not getting reassigned at any point and, as a result, don't need `global`.

Here's a couple of scenarios:
```
mydict1 = {1: 2}

def myfunction1():
    '''Assignment has no effect on mydict2 globally'''
    mydict1 = {}
    mydict1.update(x='y')

mydict2 = {3: 4}

def myfunction2():
    '''Assignment changes mydict2 globally'''
    global mydict2
    mydict2 = {}
    mydict2.update(i='j')


mydict3 = {5: 6}
def myfunction3():
    '''Update the global dictionary/no reassignment'''
    mydict3.update(i='j')


myfunction1()
myfunction2()
myfunction3()
print(mydict1)
print(mydict2)
print(mydict3)
```

```
{1: 2}
{'i': 'j'}
{5: 6, 'i': 'j'}
```

**Comment by Avasam** [comment]
I think I got confused and indeed didn't need to be global. Making the errors accurate.

**Comment by Pierre-Sassoulas** [comment]
Thanks for the investigation @mbyrnepr2 !
